repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -1918,8 +1922,8 @@ public Object getVertexAttributePreFetchCache(AtlasVertex vertex, AtlasAttribute
             return properties.get(attribute.getName());
         }
 
-        // if value is empty && element is array of primitives, return empty list
-        if (properties.get(attribute.getName()) == null && isArrayOfPrimitives) {
+        // if value is empty && element is array and not inward relation, return empty list
+        if (properties.get(attribute.getName()) == null && typeCategory.equals(TypeCategory.ARRAY) && !AtlasRelationshipEdgeDirection.IN.equals(attribute.getRelationshipEdgeDirection())) {
```

Comment:
Of the 525 attributes, I believe most fall into the `array of structs/objects` category they are OUT relations too. So they match this condition and return emptyList. This is why performance didn't degrade with adding IN also in next condition.

@mehtaanshul , please point us towards one misaligned model, we can work towards this.